wait/host_port_test.go

```diff
@@ -456,55 +463,166 @@ func TestHostPortStrategyFailsWhileInternalCheckingDueToUnexpectedContainerStatu
+
+	wg := wait.NewHostPortStrategy("80").
+		WithStartupTimeout(5 * time.Second).
+		WithPollInterval(100 * time.Millisecond)
+
+	err = wg.WaitUntilReady(context.Background(), target)
+	require.NoError(t, err)
+}
+
+func TestHostPortStrategySucceedsGivenShellIsNotFound(t *testing.T) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+	defer listener.Close()
+
+	rawPort := listener.Addr().(*net.TCPAddr).Port
+	port, err := nat.NewPort("tcp", strconv.Itoa(rawPort))
+	require.NoError(t, err)
+
+	target := &mockStrategyTarget{}
+	target.EXPECT().Host(anyContext).Return("localhost", nil)
+	target.EXPECT().Inspect(anyContext).Return(&types.ContainerJSON{
+		NetworkSettings: &types.NetworkSettings{
+			NetworkSettingsBase: types.NetworkSettingsBase{
+				Ports: nat.PortMap{
+					"80": []nat.PortBinding{
+						{
+							HostIP:   "0.0.0.0",
+							HostPort: port.Port(),
+						},
+					},
+				},
+			},
 		},
-	}
+	}, nil)
+	target.EXPECT().MappedPort(anyContext, nat.Port("80")).Return(port, nil)
+	target.EXPECT().State(anyContext).Return(&types.ContainerState{
+		Running: true,
+	}, nil)
+	target.EXPECT().Exec(anyContext, []string{"/bin/sh", "-c", "true && (\n\t\t\t\t\tcat /proc/net/tcp* | awk '{print $2}' | grep -i :0050 ||\n\t\t\t\t\tnc -vz -w 1 localhost 80 ||\n\t\t\t\t\t/bin/sh -c '</dev/tcp/localhost/80'\n\t\t\t\t)\n\t\t\t\t"}).Return(wait.ExitCmdNotFound, nil, nil)
+	target.EXPECT().CopyFileFromContainer(anyContext, testFilename).Return(nil, errNotFound)
 
-	wg := NewHostPortStrategy("80").
+	wg := wait.NewHostPortStrategy("80").
 		WithStartupTimeout(5 * time.Second).
 		WithPollInterval(100 * time.Millisecond)
 
 	err = wg.WaitUntilReady(context.Background(), target)
 	require.NoError(t, err)
 }
+
+func TestInternalCheckFailsGivenShellIsNotInstalled(t *testing.T) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+	defer listener.Close()
+
+	rawPort := listener.Addr().(*net.TCPAddr).Port
+	port, err := nat.NewPort("tcp", strconv.Itoa(rawPort))
+	require.NoError(t, err)
+
+	target := &mockStrategyTarget{}
+	target.EXPECT().Host(anyContext).Return("localhost", nil)
+	target.EXPECT().Inspect(anyContext).Return(&types.ContainerJSON{
+		NetworkSettings: &types.NetworkSettings{
+			NetworkSettingsBase: types.NetworkSettingsBase{
+				Ports: nat.PortMap{
+					"80": []nat.PortBinding{
+						{
+							HostIP:   "0.0.0.0",
+							HostPort: port.Port(),
+						},
+					},
+				},
+			},
+		},
+	}, nil)
+	target.EXPECT().MappedPort(anyContext, nat.Port("80")).Return(port, nil)
+	target.EXPECT().State(anyContext).Return(&types.ContainerState{
+		Running: true,
+	}, nil)
+	target.EXPECT().Exec(anyContext, []string{"/bin/sh", "-c", "true && (\n\t\t\t\t\tcat /proc/net/tcp* | awk '{print $2}' | grep -i :0050 ||\n\t\t\t\t\tnc -vz -w 1 localhost 80 ||\n\t\t\t\t\t/bin/sh -c '</dev/tcp/localhost/80'\n\t\t\t\t)\n\t\t\t\t"}).Return(wait.ExitEaccess, nil, nil)
```

Comment:
suggestion: you don’t need to specify the exact args when they are complex like this just use an any of type

Reply:
makes sense. I'll incorporate it with the other changes.